controllers/student/groups/groups.controller.js

```diff
@@ -0,0 +1,30 @@
+const model = require('../../../models/index')
+
+// Return all groups associated to the logged in user
+exports.findAll = (req, res) => {
+  model.Group.findAll()
+    .then(groups => {
+      res.send(groups);
+    })
+    .catch(err => {
+      res.status(500).send({
+        message:
+          err.message || "Some error occurred while retrieving groups."
+      });
+    });
+}
+
+// Return information about a single group that is associated to the logged in user
+exports.findOne = (req, res) => {
+  model.Group.findByPk(req.params.id)
+  .then(group => {
+    if (group == null) {
+      res.status(404).send({ message: `Group with id: ${req.params.id} does not exist` });
+    } else {
+      res.send(group);
+    }
+  })
+  .catch(err => {
+    res.status(500).send({ message: err.message || 'Something went wrong' });
+  })
+}
```

Comment:
Returns all groups, not just the ones the student is in, BUT in controllers/student/groups/profile/profile.controller.js , the displaying of GroupAllocation model works fine.

Student is able to get access group information of groups he's not allocated to.

Reply:
Done!